### PR TITLE
TSDB-32: Do not append if sample is too old.

### PR DESCRIPTION
**Pull request by gtopper**

### CODE REVIEW

File: pkg/chunkenc/xor.go [hunk 1]
```diff
@@ -163,6 +163,11 @@ func (a *xorAppender) Append(t int64, v float64) {
 	var tDelta uint64
 	num := *a.samples
 
+	// Do not append if sample is too old.
+	if t < a.t {
+		return
```
Comment (gshatz) [inline pkg/chunkenc/xor.go]: please add a debug print for discarding this sample

Comment (gtopper) [reply pkg/chunkenc/xor.go]: That's actually not possible without turning this PR into a logger refactoring PR.

File: pkg/chunkenc/chunkenc_test.go [hunk 2]
```diff
@@ -127,38 +130,6 @@ func TestBstream(t *testing.T) {
 
 }
 
-func DecodeTest(blob string) error {
```
Comment (gshatz) [inline pkg/chunkenc/chunkenc_test.go]: why did you delete this test?
